transactions/README.md

```diff
@@ -163,5 +241,32 @@ TODO: Summarize L15 slides 43-46
 TODO: Summarize (preferably with ASCII art) L15 slides 47-50
 
 ## Black/White Marble Example
-
-TODO: Snapshot isolation but not Serializable example (preferably with ASCII art) L16 slide 14
+Let's say we have a bag of marbles, half of which are white, and half of which are black.
+
+There are two transactions:  
+
+```
+(T1) Change all white marbles to black
+(T2) Change all black marbles to white
+```
+
+### Serializability Guarantee (2PL, OCC)
+Below are all the possibilities of schedules.
+
+```
+T1 -> T2: bag is ALL white
+T2 -> T1: bag is ALL black
+```
+
+### Snapshot Isolation (MVCC)
+Below are all the possibilities of schedules.
+
+```
+T1 -> T2: bag is ALL white
+T2 -> T1: bag is ALL black
+T1 || T2: 
```

Comment:
That's a bit of a confusing way to explain it. You're assuming snapshot reads are being realized on some real RAM memory, but I'd stick to semantics only. i.e.:

Suppose initially at snap 0 x = white marbles y = black
T1 reads snapshot 0, it will be changing x to black
T2 reads snapshot 0, it will be changing y to white
T1 writes snapshot 1, all are black
T2 writes snapshot 2, now x is black and y is white

Alternatively, T1 can read snap 0, write snap 1, and T2 reads snap 1, writes 2, causing all-white.
And then flip version of above.